src/test/java/com/ultimatelogger/multiplatform/nonmockist/logggingifisonmultiprioritylogger/TagBuildingTest.kt

```diff
@@ -0,0 +1,266 @@
+package com.ultimatelogger.multiplatform.nonmockist.logggingifisonmultiprioritylogger
+
+import com.nhaarman.mockitokotlin2.times
+import com.ultimatelogger.multiplatform.nonmockist.logggingifisonmultiprioritylogger.StubTagData.givenClassName
+import com.ultimatelogger.multiplatform.nonmockist.logggingifisonmultiprioritylogger.StubTagData.givenFileName
+import com.ultimatelogger.multiplatform.nonmockist.logggingifisonmultiprioritylogger.StubTagData.givenLineNumber
+import com.ultimatelogger.multiplatform.nonmockist.logggingifisonmultiprioritylogger.StubTagData.givenMethodName
+import com.ultimatelogger.multiplatform.nonmockist.logggingifisonmultiprioritylogger.StubTagData.givenMsg
+import com.ultimatelogger.multiplatform.tag.TagSettings
+import org.junit.Test
+import org.mockito.Mockito
+
+internal class TagBuildingTest : LoggingIfIsOnMultiPriorityLoggerTest() {
+
+    override fun setUp() {
+        stubStackTraceProvider()
+    }
+
+    @Test
+    fun `shouldLogFileNameAndLineNum = true | shouldLogClassName = true | shouldLogMethodName = true | shouldLogThreadName = true`() {
+        initLogger(
+            true,
+            TagSettings(
+                shouldLogFileNameAndLineNum = true,
+                shouldLogClassName = true,
+                shouldLogMethodName = true,
+                shouldLogThreadName = true
+            )
+        )
+
+        testVMethodCalled(
+            "[main] ($givenFileName:$givenLineNumber) $givenClassName.$givenMethodName()"
+        )
+    }
+
+    private fun testVMethodCalled(expectedTag: String) {
```

Comment:
Maybe move method declaration after the last usage?

Reply:
Rather prefer after first one.